code/edge/src/androidTest/java/com/adobe/marketing/mobile/EdgeFunctionalTests.java

```diff
@@ -537,37 +619,71 @@ public void testSendEvent_withXDMSchema_sendsExEdgeNetworkRequest() throws Excep
 			TIMEOUT_MILLIS
 		);
 		assertEquals(1, resultRequests.size());
-		Map<String, String> resultPayload = mockNetworkService.getFlattenedNetworkRequestBody(resultRequests.get(0));
-		assertEquals(17, resultPayload.size());
-
-		assertEquals("true", resultPayload.get("meta.konductorConfig.streaming.enabled"));
-		assertEquals("\u0000", resultPayload.get("meta.konductorConfig.streaming.recordSeparator"));
-		assertEquals("\n", resultPayload.get("meta.konductorConfig.streaming.lineFeed"));
-		assertNotNull(resultPayload.get("xdm.identityMap.ECID[0].id"));
-		assertEquals("false", resultPayload.get("xdm.identityMap.ECID[0].primary"));
-		assertEquals("ambiguous", resultPayload.get("xdm.identityMap.ECID[0].authenticatedState"));
-		assertNotNull(resultPayload.get("events[0].xdm._id"));
-		assertNotNull(resultPayload.get("events[0].xdm.timestamp"));
-
-		assertEquals("testWithXdmSchema", resultPayload.get("events[0].xdm.stringObject"));
-		assertEquals("100", resultPayload.get("events[0].xdm.intObject"));
-		assertEquals("true", resultPayload.get("events[0].xdm.boolObject"));
-		assertEquals("3.42", resultPayload.get("events[0].xdm.doubleObject"));
-		assertEquals("testInnerObject", resultPayload.get("events[0].xdm.xdmObject.innerKey"));
-		assertEquals("abc123def", resultPayload.get("events[0].meta.collect.datasetId"));
-
-		assertTrue(resultRequests.get(0).getUrl().startsWith(EXEDGE_INTERACT_URL_STRING));
-		assertEquals(CONFIG_ID, resultRequests.get(0).queryParam("configId"));
-		assertNotNull(resultRequests.get(0).queryParam("requestId"));
-
-		assertEquals("app", resultPayload.get("xdm.implementationDetails.environment"));
-		assertEquals(
-			EdgeJson.Event.ImplementationDetails.BASE_NAMESPACE,
-			resultPayload.get("xdm.implementationDetails.name")
-		);
-		assertEquals(
-			MobileCore.extensionVersion() + "+" + Edge.extensionVersion(),
-			resultPayload.get("xdm.implementationDetails.version")
+
+		TestableNetworkRequest networkRequest = resultRequests.get(0);
+		assertTrue(networkRequest.getUrl().startsWith(EXEDGE_INTERACT_URL_STRING));
+		assertEquals(CONFIG_ID, networkRequest.queryParam("configId"));
+		assertNotNull(networkRequest.queryParam("requestId"));
+
+		String expected =
+			"{" +
+			"  \"events\": [" +
+			"    {" +
+			"      \"meta\": {" +
+			"        \"collect\": {" +
+			"          \"datasetId\": \"abc123def\"" +
+			"        }" +
+			"      }," +
+			"      \"xdm\": {" +
+			"        \"_id\": \"STRING_TYPE\"," +
+			"        \"boolObject\": true," +
```

Comment:
nit: make it same as testString, testBool, testInt ?

Reply:
I actually mean the keys. Everywhere we have keys in format "testString", "testBool", here we have "boolObject", "doubleObject" etc...